src/test/java/org/opensearch/security/support/JsonFlattenerTest.java

```diff
@@ -0,0 +1,22 @@
+package org.opensearch.security.support;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+
+import org.junit.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+public class JsonFlattenerTest {
+    @Test
+    public void testFlattenAsMapBasic() {
+        Map<String, Object> flattenedMap = JsonFlattener.flattenAsMap("{\"key\": {\"nested\": 1}, \"another.key\": [\"one\", \"two\"] }");
```

Comment:
Ltes add more simple and complex JSON tests.

Reply:
+1. Can we add a comprehensive test suite verifying all methods of the new JsonFlattener?

Reply:
@DarshitChanpura I can add more complex tests for the overall flattenAsMap function but I don't think making tests for the other functions is needed/would work since there is no return value and every other function is used within the workflow of flattenAsMap. Is there a particular benefit to writing smaller tests for these helper functions?